src/main/java/org/haffson/adventofcode/days/day07/Step.java

```diff
@@ -0,0 +1,17 @@
+package org.haffson.adventofcode.days.day07;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public record Step(char stepName, @Nonnull List<Character> dependsOn) {
```

Comment:
Wenn `dependsOn` ein Set wäre, dann könnte man Duplikate in den Dependencies direkt ausschließen.

Reply:
Interessant. Dadurch lassen sich definitiv weitere Fehlerquellen ausschließen.
Bei der aktuellen Gestaltung konnte es zu keinen Duplikaten kommen, aber bei anderen Anwendungsfällen durchaus sinnvoll.
(Die einzigen Duplikate kamen beim Zusammenbau einer Hilfsmethode zustande.
Diese hatte aber keine doppelten `dependsOn`, sondern doppelte `stepName` mit verschiedenen `dependsOn` )